lib/actions/errorDialogActions.js

```diff
@@ -39,8 +39,20 @@
 export const ERROR_DIALOG_SHOW = 'ERROR_DIALOG_SHOW';
 
 export function showErrorDialog(error) {
+    if (!error) {
+        return {
+            type: ERROR_DIALOG_SHOW,
+            message: 'Undefined error',
+        };
+    }
+    if (typeof error === 'string') {
+        return {
+            type: ERROR_DIALOG_SHOW,
+            message: error,
+        };
+    }
     return {
         type: ERROR_DIALOG_SHOW,
-        message: error.message || error,
+        message: error.message,
```

Comment:
Is this change done only to handle the case that error was sometimes `undefined`? If so, then this could be easier solved with a default parameter:
```js
export function showErrorDialog(error = 'Undefined error') {
    return {
        type: ERROR_DIALOG_SHOW,
        message: error.message || error,
    };
}
```

If you really want to also catch other cases, (e.g. error could also be `null`), then I think it would be nicer to create a function to just extract the right message out of `error`:

```js
const extractMessage = error => {
    if (error && error.message) {
        return error.message;
    }

    if (typeof error === 'string') {
        return error;
    }

    return 'Undefined error';
};

export const showErrorDialog = error => ({
    type: ERROR_DIALOG_SHOW,
    message: extractMessage(error),
});
```

Reply:
You are quite right, update coming up.